src/tools/kata-ctl/src/iptables.rs

```diff
@@ -0,0 +1,153 @@
+use clap::{App, Arg, SubCommand};
+use reqwest::{Url};
+use std::{fs, process};
+use shimclient::MgmtClient;
+
+//kata-proxy management API endpoint, without code would not know the location of the unix sockets
+const DEFAULT_TIMEOUT: u64 = 30;
+const IP_TABLES_SOCKET: &str = "unix:///run/vc/sbs/{sandbox_id}/ip_tables";
+const IP6_TABLES_SOCKET: &str = "unix:///run/vc/sbs/{sandbox_id}/ip6_tables";
+
+//main function for error handeling
+fn main() {
+    if let Err(e) = new_main() {
+        eprintln!("Error: {}", e);
+        process::exit(1);
+    }
+}
+
+// Define the function signature. It returns `Result<(), Box<dyn std::error::Error>>`,
+// which means it can either return an `Ok(())` value indicating success or an `Err` value
+// containing a boxed error type that implements the `std::error::Error` trait.
+fn new_main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
You don't need to create any main functions as that's already handled by https://github.com/kata-containers/kata-containers/blob/main/src/tools/kata-ctl/src/main.rs.

What you need to do is implement the `handle_iptables()` function called here:

https://github.com/kata-containers/kata-containers/blob/main/src/tools/kata-ctl/src/main.rs#L34

See my comment here: https://github.com/kata-containers/kata-containers/issues/5343#issuecomment-1446709040.

You'll also need to add the command-line options/args, but again, only by adding to what is already in `src/main.rs` (see the `IptablesCommand` structure).

What I would do though is move the existing "stub" / "nop" `handle_iptables()` function from `src/ops/check_ops.rs` into your new `src/iptables.rs` file (it got added to the `check_ops.rs` file originally by mistake I think ;)

You'll also need to add `mod iptables;` to `src/main.rs` as currently this file is _not_ being built as part of `kata-ctl` as it isn't being referenced by any of the existing code (and in fact fails to compile due to the "floating match" (see comment below).